### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by lincoln1883**
For this project check-point I have completed the following requirements:
- [x] Add the following validations:
  - [x] For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - [x] For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- [x] Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,15 +3,11 @@ class User < ApplicationRecord
...
 
-  after_commit :update_posts_counter, if: :saved_change_to_id?
+  validates :name, presence: true, length: { minimum: 3 }
+  validates :posts_counter, numericality: { integer: true, greater_than_or_equal_to: 0 }
+  validates :bio, presence: true, length: { minimum: 5, maximum: 500 }
 
   def recent_posts(limit = 3)
     posts.order(created_at: :desc).limit(limit)
   end
-
```
Comment (nahnahdev) [inline app/models/user.rb:11]: - [ ] Great job so far @lincoln1883 There are just a few changes I will suggest you implement and you'll be on your way to the next milestone :+1: 
- [ ] Per the project requirements, you were asked to `Add unit specs for all of your models' methods and validations` You have done an amazing job adding most of the tests. I noticed that this `recent_posts` method hasn't been tested yet. So kindly include that in your tests :wink:

File: app/models/post.rb [hunk 2]
```diff
@@ -1,29 +1,22 @@
...
+  validates :comments_counter, numericality: { integer: true, greater_than_or_equal_to: 0 }
 
-  def update_counters
-    update(comments_counter: comments.count, likes_counter: likes.count)
-  end
+  after_create :increment_posts_counter
 
   def recent_comments(limit = 5)
     comments.order(created_at: :desc).limit(limit)
   end
```
Comment (nahnahdev) [inline app/models/post.rb:15]: - [ ] Kindly add `unit specs` for this `recent_comments` method as well :+1:

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @lincoln1883 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights :fire: 
- All required user validations added :heavy_check_mark: 
- All required posts validations added :heavy_check_mark: 
- Has unit tests for `User`, `Post`, `Comment`, and `Like` model :heavy_check_mark: 
- All tests passing :heavy_check_mark: 
- Model specs has **Validation** tests :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DuaneDave — APPROVED** [review]
Hi @lincoln1883 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ All custom methods in the models are covered in your specs

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**